components/link/TopNavbarLink.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Icon } from 'react-components';
+import { Link } from 'react-router-dom';
+
+const TopNavbarLink = ({ to, icon, text, ...rest }) => {
+    const iconComponent =
+        typeof icon === 'string' ? (
+            <Icon className="topnav-icon mr0-5 flex-item-centered-vert fill-white" name={icon} />
+        ) : (
+            icon
+        );
+
+    return (
+        <Link to={to} {...rest}>
+            {iconComponent}
+            <span className="navigation-title topnav-linkText">{text}</span>
+        </Link>
+    );
+};
+
+TopNavbarLink.propTypes = {
+    to: PropTypes.string.isRequired,
+    icon: PropTypes.string.isRequired,
```

Comment:
Why not using `node` here? I don't understand why we need to handle node and string with `iconComponent`.

Reply:
I just wanted to imitate what we have in the `Button` component, where `icon` can be either a string (name of the icon) or a node (icon itself). But I forgot to modify the PropType accordingly. I did that with the last push.

Reply:
I don't see the point of using both, just use `PropTypes.string` please.